js/components/dashboard/expand-students.js

```diff
@@ -0,0 +1,28 @@
+import React, { PureComponent } from 'react'
+import { Map, List } from 'immutable'
+import { SET_STUDENTS_EXPANDED } from '../../actions/dashboard'
+import Button from '../report/button'
+
+import css from '../../../css/dashboard/expand-students.less'
+
+export default class ExpandStudents extends PureComponent {
+  onSetStudentsExpanded (value) {
+    const { setStudentsExpanded, students } = this.props
+    setStudentsExpanded(students.map(student => student.get('id')), value)
+  }
+
+  render () {
+    const { expandedStudents } = this.props
+    const anyStudentsExpanded = expandedStudents.some((isExpanded) => isExpanded)
+    return (
+      <Button onClick={() => this.onSetStudentsExpanded(!anyStudentsExpanded)} className={css.expandStudents}>
```

Comment:
I know we talked about not passing arrow functions to React components...but it looks like the `Button` only re-renders when clicked, and the behavior is also here:

https://github.com/concord-consortium/portal-report/blob/master/js/components/report/report.js#L84

Reply:
You're definitely right, it doesn't matter here, Button doesn't have any children and expensive re-rendering. But I'd avoid it just to have a consistent style. And avoid a situation where someone copies that pattern in place where it actually matters. I know you pointed out some existing inconsistency, most likely added by me. :wink: But still, I wouldn't add new one.

If you don't like adding constructor just for that, you could also use method trick proposed by Kirk:
```
onClick = () => {
  ...
}
```
(note there's arrow function)
that provides auto-binding too. I'd vote for constructor as that's the main style in this project. Honestly, I never care about particular style (like here), but I do care more about consistency and common style. Most of CC project don't have that and it's painful for me. :wink:

Reply:
Alright, I took care of this! I originally tried @kswenson's arrow function, but I realized I couldn't get it to work until I added the proper Babel transformer. This still didn't solve the problem (also in the code I referenced) of wanting to pass different parameters to the click handler in different cases. Binding them in the render function is just as bad as before, but I found this neat recommendation!

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md#protips

I ended up creating a sub-component for the button, so that I can pass different values to the click handler without binding functions in `props`.

Reply:
As for a lack of consistency across CC projects...I don't want you to be in pain, but I think I do like the arrow functions over binding in the constructor! So I used them in the updated sub-component 😬